Tue Dec  7 16:39:35 EST 2010  "Roland Haas" <roland.haas@physics.gatech.edu>
  * visitCarpetHDF5: set up nesting information for 2d data correctly
diff -rN -u old-tools/visitCarpetHDF5/avtCarpetHDF5FileFormat.C new-tools/visitCarpetHDF5/avtCarpetHDF5FileFormat.C
--- old-tools/visitCarpetHDF5/avtCarpetHDF5FileFormat.C	2011-07-12 09:18:51.000000000 -0400
+++ new-tools/visitCarpetHDF5/avtCarpetHDF5FileFormat.C	2011-07-12 09:18:51.000000000 -0400
@@ -307,6 +307,10 @@
             float y1 = comp[0][i].ymax();
             float z0 = comp[0][i].zmin();
             float z1 = comp[0][i].zmax();
+
+	    bool ghost = true;
+    	    dims = comp[0][i].npoints(ghost);
+	    
             for (int j = 0; j < num_patches; j++)
             {
                 if (comp[0][j].rl() != comp[0][i].rl()+1)
@@ -319,17 +323,19 @@
                 float c0 = comp[0][j].zmin();
                 float c1 = comp[0][j].zmax();
 
-                if (a0 >= x1 || x0 >= a1 ||
-                    b0 >= y1 || y0 >= b1 ||
-                    ((num_dims == 3) && (c0 >= z1 || z0 >= c1)))
+                float dx = comp[0][j].delta()[0]/2;
+                float dy = comp[0][j].delta()[1]/2;
+                float dz = comp[0][j].delta()[2]/2;
+
+                // compare with a half-finer level offset to work around floating point roundoff issues
+                if (((dims[0] > 1) && (a0 >= x1-dx || x0 >= a1-dx)) ||
+                    ((dims[1] > 1) && (b0 >= y1-dy || y0 >= b1-dy)) ||
+                    ((dims[2] > 1) && (c0 >= z1-dz || z0 >= c1-dz)))
                     continue;
 
                 childPatches.push_back(j);
             }
 
-	    bool ghost = true;
-    	    dims = comp[0][i].npoints(ghost);
-	    
 	    //bool isOutermostX = comp[0][i].is_at_upper_total_rl_domain_boundary(0, data_file->timesteps[timeState].totalCartDomainIExt[comp[0][i].rl()][1]);
     	    //bool isOutermostY = comp[0][i].is_at_upper_total_rl_domain_boundary(1, data_file->timesteps[timeState].totalCartDomainIExt[comp[0][i].rl()][3]);
     	    //bool isOutermostZ = comp[0][i].is_at_upper_total_rl_domain_boundary(2, data_file->timesteps[timeState].totalCartDomainIExt[comp[0][i].rl()][5]);
